protocol.js

```diff
@@ -34,11 +34,11 @@ class Protocol extends Duplex {
     this.pending = new Map()
 
     if (opts && 'function' === typeof opts.connect) {
-      process.nextTick(pump, this, opts.connect(this), this)
+      process.nextTick(pump, opts.connect(this), this, opts.connect(this))
```

Comment:
Nice! We should store the result of opts.connect() and use it instead of calling is twice in the event that the function is a factory returning a new stream each time

Reply:
Oh yeah good point.

Reply:
Done